src/backend.rs

```diff
@@ -220,14 +223,31 @@ pub trait Backend {
     ///
     /// ```rust
     /// # use ratatui::backend::{Backend, TestBackend};
+    /// # use ratatui::layout::Position;
     /// # let mut backend = TestBackend::new(80, 25);
-    /// backend.set_cursor(10, 20)?;
-    /// assert_eq!(backend.get_cursor()?, (10, 20));
+    /// backend.set_cursor_position(Position { x: 10, y: 20 })?;
+    /// assert_eq!(backend.get_cursor_position()?, Position { x: 10, y: 20 });
     /// # std::io::Result::Ok(())
     /// ```
+    fn set_cursor_position<P: Into<Position>>(&mut self, position: P) -> io::Result<()>;
+
+    /// Get the current cursor position on the terminal screen.
     ///
-    /// [`get_cursor`]: Backend::get_cursor
-    fn set_cursor(&mut self, x: u16, y: u16) -> io::Result<()>;
```

Comment:
Aside (no action necessary just an FYI): Something that can make diffs like this easier to read (on GitHub at least) is to make additive changes like this below the original implementation that stays. E.g. see your change on Terminal below which puts the new method below the old one. I'm guessing here it made sense to put it above because we're grouping provided methods below required methods?

Reply:
In this case, the diff is kind of horrible in any order as almost all lines contain changes, and it finds empty doc string lines (`///`) to match at random points resulting in a weird diff.